src/pim-types.ts

```diff
@@ -255,6 +264,57 @@ export class TaskType extends EventType {
     ret.color = this.color;
     return ret;
   }
+
+  public getNextOccurence(): TaskType | null {
+    if (!this.isRecurring()) {
+      return null;
+    }
+
+    const rrule = this.rrule.clone();
+
+    if (rrule.count && rrule.count <= 1) {
+      return null; // end of reccurence
+    }
+
+    rrule.count = null; // clear count so we can iterate as many times as needed
+    const recur = rrule.iterator(this.startDate ?? this.dueDate);
+    let next = recur.next();
+    while ((next = recur.next())) {
+      if (next.compare(ICAL.Time.now()) > 0) {
+        break;
+      }
+    }
+
+    if (!next) {
+      return null; // end of reccurence
+    }
+
+    const nextStartDate = this.startDate && next;
+    const nextDueDate = this.dueDate && next;
+    if (nextStartDate && nextDueDate) {
+      const offset = (this.dueDate as ICAL.Time).subtractDateTz(this.startDate);
+      nextDueDate.addDuration(offset);
+    }
+
+    const nextTask = this.clone();
+    nextTask.uid = uuid.v4();
+    if (nextStartDate) {
+      nextTask.startDate = nextStartDate;
+    }
+    if (nextDueDate) {
+      nextTask.dueDate = nextDueDate;
+    }
+
+    if (this.rrule.isByCount()) {
+      rrule.count = (this.rrule.count as number) - 1;
```

Comment:
Why is this cast needed?

Reply:
`rrule.count` can be `null`. `rrule.isByCount` actually checks if this number is `null`, but Typescript doesn't know that so it complains. I guess I could just change the check to use `rrule.count` instead of `rrule.isByCount`.

Reply:
the correct way to count optional to non-optional in typescript is the `!` operator. So `this.rrule.count! - 1`.

Alternatively, why not just check if `this.rrule.count` is not null instead of using `isByCount`? I'm sure there's a reason, as they created this function, I'm just wondering.

Reply:
> Alternatively, why not just check if this.rrule.count is not null instead of using isByCount? I'm sure there's a reason, as they created this function, I'm just wondering.

Yeah I was just trying to be semantic with my use of the library. Actually, looking at the source code the `isByCount` function does more than I need it to. It `count` and `until`, which I don't want. So I'll just use `this.rrule.count`.

Reply:
Great.